src/internal/operators/catchError.ts

```diff
@@ -153,3 +151,21 @@ export function catchError<T, O extends ObservableInput<any>>(
       return subscription;
     });
 }
+
+/**
+ * This must exist to ensure that the `closed` state of the inner subscriber is set at
+ * the proper time to ensure operators like `take` can stop the inner subscription if
+ * it is a synchronous firehose.
+ */
+class CatchErrorSubscriber<T, C> extends Subscriber<T> {
+  ready = false;
```

Comment:
issue: I cannot see where this property is used. Do we really need it?

Reply:
oops. That's cruft.